File: C# - Graph Searching/BFS.cs
using System;
using System.Collections;
using System.Collections.Generic;

namespace Przeszukiwanie_Grafow{
    internal static class Bfs{
        internal static void Traverse(int starting, List<List<int>> adjlist){
            Console.WriteLine("BFS: ");
            var queue=new Queue();
            var visited=new ArrayList{adjlist[starting][0]};
            for(var i=1;i<adjlist[starting].Count;i++)
                if(!visited.Contains(adjlist[starting][i]))
                    queue.Enqueue(adjlist[starting][i]);
            while(queue.Count>0){
                var v=(int)queue.Peek();
                queue.Dequeue();
                visited.Add(v);
                for(var i=1;i<adjlist[v].Count;i++)
                    if(!queue.Contains(adjlist[v][i]))
                        if(!visited.Contains(adjlist[v][i]))
                            queue.Enqueue(adjlist[v][i]);
            }
            foreach(var visit in visited) Console.Write(visit+" ");
        }
    }
}

File: Python - Genetic Algo Labirynth/bfs.py
from collections import deque


def bfs(lab):
    start = (1, 1)
    end = (len(lab) - 2, len(lab) - 2)
    path = []
    queue = [start]
    visited = [start]
    while queue:
        (x, y) = queue.pop(0)
        path.append((x, y))
        if (x, y) == end:
            return path
        if (x + 1, y) not in visited:
            visited.append((x + 1, y))
            if(lab[y][x + 1]):
                queue.append((x + 1, y))
        if (x - 1, y) not in visited:
            visited.append((x - 1, y))
            if(lab[y][x - 1]):
                queue.append((x - 1, y))
        if (x, y + 1) not in visited:
            visited.append((x, y + 1))
            if(lab[y + 1][x]):
                queue.append((x, y + 1))
        if (x, y - 1) not in visited:
            visited.append((x, y - 1))
            if(lab[y - 1][x]):
                queue.append((x, y - 1))
    print("no path")
